File: XRViewer/ARKController/SceneKit/SCNNode+Show.m
#import "SCNNode+Show.h"

@implementation SCNNode (Show)

- (void)show:(BOOL)show
{
    if (show)
    {
        [self unhide];
    }
    else
    {
        [self hide];
    }
}

- (void)hide
{
    if ([self opacity] == 1.0)
    {
        [self runAction:[SCNAction fadeOutWithDuration:0.5]];
    }
}

- (void)unhide
{
    if ([self opacity] == 0.0)
    {
        [self runAction:[SCNAction fadeInWithDuration:0.5]];
    }
}

@end

File: XRViewer/ARKController/SceneKit/SCNNode+Show.swift
import SceneKit

extension SCNNode {
    func show(_ show: Bool) {
        if show {
            unhide()
        } else {
            hide()
        }
    }

    func hide() {
        if opacity == 1.0 {
            runAction(SCNAction.fadeOut(duration: 0.5))
        }
    }

    func unhide() {
        if opacity == 0.0 {
            runAction(SCNAction.fadeIn(duration: 0.5))
        }
    }
}
